benchmark/run_all.dart

```diff
@@ -24,27 +26,46 @@ List<int> throughputParserHelper(String aotThroughputStdout) {
 }
 
 void main(List<String> args) async {
-  var url = 'https://example.com';
-  var throughputPrallelLimit = 10;
-  var duration = const Duration(seconds: 1);
-  if (args.isNotEmpty) {
-    url = args[0];
-  }
-  if (args.length > 1) {
-    throughputPrallelLimit = int.parse(args[1]).toInt();
+  final parser = ArgParser();
+  parser
+    ..addOption('url',
+        abbr: 'u',
+        help: 'The server to ping for running this benchmark.',
+        defaultsTo: 'https://example.com')
+    ..addOption('limit',
+        abbr: 'l',
+        help: 'Limits the maximum number of parallel requests to 2^N where N'
+            ' is provided through this option.',
+        defaultsTo: '10')
+    ..addOption('time',
+        abbr: 't',
+        help: 'Maximum second(s) the benchmark should wait for each request.',
+        defaultsTo: '1')
+    ..addFlag('help',
+        abbr: 'h', negatable: false, help: 'Print this usage information.');
+  final arguments = parser.parse(args);
+  if (arguments.wasParsed('help')) {
+    print(parser.usage);
+    return;
   }
-  if (args.length > 2) {
-    duration = Duration(seconds: int.parse(args[2]));
+  if (arguments.rest.isNotEmpty) {
+    print(parser.usage);
+    throw ArgumentError();
   }
+  final url = arguments['url'] as String;
+  final throughputPrallelLimit = int.parse(arguments['limit'] as String);
+  final duration = Duration(seconds: int.parse(arguments['time'] as String));
+
   print('Latency Test against: $url');
   print('JIT');
   final jitCronetLatency = await CronetLatencyBenchmark.main(url);
   final jitDartIOLatency = await DartIOLatencyBenchmark.main(url);
 
   print('AOT');
   print('Compiling...');
-  Process.runSync('dart', ['compile', 'exe', 'benchmarks/latency.dart']);
-  final aotLantencyProc = await Process.start('benchmarks/latency.exe', [url]);
+  Process.runSync('dart', ['compile', 'exe', 'benchmark/latency.dart']);
+  final aotLantencyProc =
+      await Process.start('benchmark/latency.exe', ['-u', url]);
```

Comment:
Do the 'exe' files compiled by dart only work on windows?

Reply:
No :sweat_smile:. It is named as `.exe` by default regardless of the platform.

Reply:
Ah, yes. I see it now in the documentation. Dart supports quite a few different compilation formats 😀